client/jetpack-cloud/index.js

```diff
@@ -32,13 +34,19 @@ const redirectToPrimarySiteLanding = ( context, next ) => {
 	const state = context.store.getState();
 	const currentUser = getCurrentUser( state );
 	const isPrimarySiteJetpackSite = getPrimarySiteIsJetpack( state );
-	context.primary = (
-		<SiteLanding
-			primarySiteSlug={ currentUser.primarySiteSlug }
-			isPrimarySiteJetpackSite={ isPrimarySiteJetpackSite }
-		/>
-	);
-	next();
+	const isAgency = isAgencyUser( state );
+	const isAgencyEnabled = config.isEnabled( 'jetpack/agency-dashboard' );
+	const dashboardRedirectLink = jetpackDashboardRedirectLink( state );
+
+	if ( isAgency && isAgencyEnabled ) {
+		page.redirect( dashboardRedirectLink );
```

Comment:
Is this `next();` necessary here since we are redirecting to the dashboard? 

I think we can just do `return page.redirect( dashboardRedirectLink );` here